### ISSUE TITLE
Empty Search Value in Combobox When Selecting 'Created At' Item

### ISSUE THREAD

**Issue by ChoaibMouhrach** [shadcn-ui/ui#943]
I was attempting to create a combobox for setting sorting on my dashboard products page. Then, I noticed that after selecting the items, which are:

```ts
const items: ComboxUiItem[] = [
  {
    label: "Id",
    value: "id",
  },
  {
    label: "Name",
    value: "name",
  },
  {
    label: "Price",
    value: "price",
  },
  {
    label: "Quantity",
    value: "quantity",
  },
  {
    label: "Created At",
    value: "createdAt"
  },
]
```

The combobox component:

```ts
import {
  Command,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
} from "@/components/ui/command"
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from "@/components/ui/popover"
import { Button } from "@/components/ui/button"
import { cn } from "@/lib/utils"
import * as React from "react"
import { CaretSortIcon, CheckIcon } from "@radix-ui/react-icons"

export interface ComboxUiItem {
  value: string;
  label: string;
}

interface ComboxUiProps {
  placeholder: string;
  items: ComboxUiItem[];
}

export function ComboxUi({ placeholder, items }: ComboxUiProps) {
  const [open, setOpen] = React.useState(false)
  const [value, setValue] = React.useState("")

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button
          variant="outline"
          role="combobox"
          aria-expanded={open}
          className="w-[200px] justify-between"
        >
          {value ? items.find((item) => item.value === value)?.label : placeholder}
          <CaretSortIcon className="ml-2 h-4 w-4 shrink-0 opacity-50" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[200px] p-0">
        <Command>
          <CommandInput placeholder={placeholder} className="h-9" />
          <CommandEmpty>Not found.</CommandEmpty>
          <CommandGroup>
            {items.map((item) => (
              <CommandItem
                key={item.value}
                onSelect={(currentValue) => {
                  setValue(currentValue === value ? "" : currentValue)
                  setOpen(false)
                }}
              >
                {item.label}
                <CheckIcon
                  className={cn(
                    "ml-auto h-4 w-4",
                    value === item.value ? "opacity-100" : "opacity-0"
                  )}
                />
              </CommandItem>
            ))}
          </CommandGroup>
        </Command>
      </PopoverContent>
    </Popover>
  )
}
```

I noticed that when selecting the "Created At" item, the combobox search value becomes empty, as shown here:

![shadcncombox](https://github.com/shadcn/ui/assets/107870395/a9006035-d8e3-4482-97d7-eeba34e2e6cc)

To fix this issue, I suggest replacing this line:

```ts
setValue(currentValue === value ? "" : currentValue)
```

with this line:

```ts
setValue(currentValue === value ? "" : items.find(item => item.label.toLocaleLowerCase() === currentValue)!.value)
```

**Comment by servesh-chaturvedi** [comment]
Hi, can you check your code with this [implementation](https://github.com/shadcn/ui/blob/main/apps/www/app/examples/tasks/). The `columns.tsx` file needs special attention.

**Comment by miquelvir** [comment]
This is caused by missing value on the CommandItem. The example was fixed in #1788, you can close this now

**Comment by shadcn** [comment]
This issue has been automatically closed because it received no activity for a while. If you think it was closed by accident, please leave a comment. Thank you.
